tests/bootstrap/test_cosmetic_configuration.py

```diff
@@ -0,0 +1,297 @@
+
+            infra.add_custom_setup(
+                lambda: load_python_lambda_to_s3(
+                    s3_client=aws_client.s3,
+                    bucket_name=infra.get_asset_bucket(),
+                    key_name=key_name,
+                    code_path=resources_path,
+                    additional_python_packages=additional_packages or [],
+                )
+            )
+
+            given_environment = environment or {}
+            base_environment = {"CUSTOM_LOCALSTACK_HOSTNAME": chosen_localstack_host}
+            full_environment = {**base_environment, **given_environment}
+            return cdk.aws_lambda.Function(
+                stack,
+                resource_name,
+                handler="index.handler",
+                code=cdk.aws_lambda.S3Code(bucket=asset_bucket, key=key_name),
+                runtime=runtime,
+                environment=full_environment,
+                **kwargs,
+            )
+
+        # SQS queue
+        queue = cdk.aws_sqs.Queue(stack, "Queue")
+
+        # SNS topic
+        topic = cdk.aws_sns.Topic(stack, "Topic")
+        topic.add_subscription(cdk.aws_sns_subscriptions.SqsSubscription(queue))
+
+        # API Gateway
+        asset_bucket = cdk.aws_s3.Bucket.from_bucket_name(
+            stack,
+            "BucketName",
+            bucket_name=infra.get_asset_bucket(),
+        )
+        apigw_handler_fn = create_lambda_function(
+            stack,
+            resource_name="ApiHandlerFn",
+            resources_path=os.path.join(os.path.dirname(__file__), "resources/apigw_handler.py"),
+            environment={
+                "TOPIC_ARN": topic.topic_arn,
+            },
+        )
+
+        api = cdk.aws_apigateway.RestApi(stack, "RestApi")
+        upload_url_resource = api.root.add_resource("upload")
+        upload_url_resource.add_method(
+            "POST", cdk.aws_apigateway.LambdaIntegration(apigw_handler_fn)
+        )
+        cdk.CfnOutput(stack, "ApiUrl", value=api.url)
+
+        # event handler lambda
+        create_lambda_function(
+            stack,
+            resource_name="EventHandlerFn",
+            resources_path=os.path.join(os.path.dirname(__file__), "resources/event_handler.py"),
+            additional_packages=["requests", "boto3"],
+            events=[
+                cdk.aws_lambda_event_sources.SqsEventSource(queue),
+            ],
+            environment={
+                "DOMAIN_ENDPOINT": domain.domain_endpoint,
+                "RESULTS_BUCKET": results_bucket.bucket_name,
+                "RESULTS_KEY": RESULT_KEY,
+            },
+        )
+
+        with infra.provisioner() as prov:
+            yield prov
+
+    def test_scenario(self, port, infrastructure, aws_client_factory, chosen_localstack_host):
+        """
+        Scenario:
+            * API Gateway handles web request
+            * Broadcasts message onto SNS topic
+            * Lambda subscribes via SQS and inserts message into OpenSearch
```

Comment:
It does not do that, right? It just checks if it's ready and puts the message into an s3 bucket.

Reply:
Absolutely correct, it was planned, but was more complex than I really wanted to get into.